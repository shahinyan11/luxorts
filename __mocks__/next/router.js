import * as Router from 'next/router';

export const router = {
  push: jest.fn(),
  back: jest.fn(),
  replace: jest.fn(),
  prefetch: jest.fn(),
  query: {},
  pathname: '/',
  beforePopState: jest.fn(),
};

export const useRouter = jest.fn(() => router);
export const { withRouter } = Router;
