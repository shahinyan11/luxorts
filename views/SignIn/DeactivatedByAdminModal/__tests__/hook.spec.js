import { renderHook } from '@testing-library/react-hooks';

import ROUTES from 'constants/routes';

import { router } from '__mocks__/next/router';
import useContainer from '../hook';

describe('DeactivatedByAdminModal useContainer hook', () => {
  const onClose = jest.fn();
  const { result } = renderHook(() => useContainer({ onClose }));

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot();
  });

  describe('tests `onConfirmHandler` method', () => {
    beforeEach(() => {
      result.current.onConfirmHandler();
    });

    it('should calls router push', () => {
      expect(router.push).toHaveBeenCalledWith(ROUTES.SUPPORT.PATH);
    });

    it('should calls onClose callback', () => {
      expect(onClose).toHaveBeenCalled();
    });
  });
});
