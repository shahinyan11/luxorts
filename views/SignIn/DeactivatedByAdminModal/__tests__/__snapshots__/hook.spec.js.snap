// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeactivatedByAdminModal useContainer hook matches snapshot 1`] = `
Object {
  "onConfirmHandler": [Function],
}
`;
